### PR TITLE
[ARM] Added instructions for resolving special characters in PowerShell

### LINKED ISSUE #10907: Typo in group deployment command when defining parameters
In the following command 
`az group deployment create -g MyResourceGroup --template-uri https://myresource/azuredeploy.json --parameters @myparameters.json`

`--parameters @myparameters.json` part should be `--parameters '@myparameters.json'`

Tested on PowerShell Core
---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: de233487-a2e5-00ab-30e9-f68866735957
* Version Independent ID: 2cd3ed9f-113f-a6c2-6a36-2925a044edf8
* Content: [az group deployment](https://docs.microsoft.com/en-us/cli/azure/group/deployment?view=azure-cli-latest)
* Content Source: [src/azure-cli/azure/cli/command_modules/resource/_help.py](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/resource/_help.py)
* Service: **azure-resource-manager**
* GitHub Login: @rloutlaw
* Microsoft Alias: **routlaw**

### PR DESCRIPTION
**Pull request by zhoxing-ms**
Fix #10907 
Added instructions for resolving special characters in PowerShell

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: doc/use_cli_effectively.md [hunk 1]
```diff
@@ -192,6 +192,13 @@ To prevent this, you may use [stop-parsing symbol `--%`](https://docs.microsoft.
 
 This issue is tracked at https://github.com/PowerShell/PowerShell/issues/1995#issuecomment-539822061
 
+### Special characters in PowerShell affect parameter parsing
+When cli is used in power shell, parsing errors may occur when special characters of PowerShell are entered as parameters.
```
Comment (Juliehzl) [inline doc/use_cli_effectively.md]: power shell -> PowerShell

Comment (zhoxing-ms) [reply doc/use_cli_effectively.md]: Done

**Comment by jiasli** [comment]
Please fix the escaping of `` ` `` correctly.

File: doc/use_cli_effectively.md [hunk 2]
```diff
@@ -182,9 +182,16 @@ foreach ($vm_id in $vm_ids) {
...
+
+#### Quotes get stripped by PowerShell
+On Windows, `az` is a CMD script (at `C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\wbin\az.cmd`). When there is no space in an argument, PowerShell will strip the quotes and pass the argument to CMD. This causes the argument to be parsed again by CMD. For example, when running `az "a&b"` in PowerShell, `b` is treated as a separate command instead of part of the argument like CMD does, because quotes are removed by PowerShell and the ampersand `&` is parsed again by CMD as a [command separator](https://docs.microsoft.com/en-us/previous-versions/windows/it-pro/windows-xp/bb490954(v=technet.10)#using-multiple-commands-and-conditional-processing-symbols). 
+
+To prevent this, you may use [stop-parsing symbol `--%`](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_parsing) between `az` and arguments like `az --% vm create ...`. This can also solve the above-mentioned special character issue and is recommended whenever any issue happens when invoking a CMD batch script from PowerShell.
```
Comment (qwordy) [inline doc/use_cli_effectively.md]: Maybe we can move `--%` outside the link to make the page rendered look better.

File: doc/use_cli_effectively.md [hunk 3]
```diff
@@ -182,9 +182,16 @@ foreach ($vm_id in $vm_ids) {
...
+
+#### Quotes get stripped by PowerShell
+On Windows, `az` is a CMD script (at `C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\wbin\az.cmd`). When there is no space in an argument, PowerShell will strip the quotes and pass the argument to CMD. This causes the argument to be parsed again by CMD. For example, when running `az "a&b"` in PowerShell, `b` is treated as a separate command instead of part of the argument like CMD does, because quotes are removed by PowerShell and the ampersand `&` is parsed again by CMD as a [command separator](https://docs.microsoft.com/en-us/previous-versions/windows/it-pro/windows-xp/bb490954(v=technet.10)#using-multiple-commands-and-conditional-processing-symbols).
```
Comment (qwordy) [inline doc/use_cli_effectively.md]: Command Prompt is the official app name of CMD

File: doc/use_cli_effectively.md [hunk 4]
```diff
@@ -115,6 +115,16 @@ This becomes an issue because when the command shell (bash, zsh, Windows command
     - `--arg foo bar`: OK. Unquoted, space-separated list
     - `--arg "foo" "bar"`: OK: Quoted, space-separated list
     - `--arg "foo bar"`: BAD. This is a string with a space in it, not a space-separated list.
+9. When running Azure CLI commands in PowerShell, parsing errors will occur when the arguments contain special characters of PowerShell, such as at `@`. You can solve this problem by adding `` ` `` before the special character to escape it, or by enclosing the argument with single or double quotes `'`/`"`. For example, `az group deployment create --parameters @parameters.json` dose't work in PowerShell because `@` is parsed as a [splatting symbol](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_splatting). To fix this, you may change the argument to `` `@parameters.json`` or `'@parameters.json'`.
```
Comment (qwordy) [inline doc/use_cli_effectively.md]: `'` or `"`

Comment (jiasli) [reply doc/use_cli_effectively.md]: This is good IMO.

**Pull request merged**